pkg/testutils/testutils.go

```diff
@@ -138,17 +139,21 @@ func EnsureServiceHealthy(t testing.TB, grpcAddr, httpAddr string, transportCred
 	}
 
 	dialOpts := []grpc.DialOption{
-		grpc.WithBlock(),
 		grpc.WithTransportCredentials(creds),
 		grpc.WithConnectParams(grpc.ConnectParams{Backoff: grpcbackoff.DefaultConfig}),
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
 	conn, err := grpc.DialContext(
-		context.Background(),
+		ctx,
 		grpcAddr,
 		dialOpts...,
 	)
-	require.NoError(t, err)
+	if err != nil {
```

Comment:
It's not obvious what the purpose of this change and adding the `error` signature to the return of this function is. Could you elaborate?

This causes us to have to do

```go
err := testutils.EnsureServiceHealthy(t, cfg.GRPC.Addr, cfg.HTTP.Addr, nil, true)
require.NoError(t, err)
```

everywhere in our test code. Whereas before we get the `require.NoError(..)` right here inline. Every test that used `EnsureServiceHealth` got that for free without having to check the returned error.